pkg/adapter/kubernetes/config/health.go

```diff
@@ -0,0 +1,154 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	k8sErr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/util"
+)
+
+const (
+	healthConfigMapName      = "k8s-blueprint-operator-health-config"
+	componentHealthConfigKey = "components"
+	waitHealthConfigKey      = "wait"
+)
+
+var defaultHealthConfig = healthConfig{
+	Components: componentHealthConfig{
+		Required: nil,
+	},
+	Wait: waitHealthConfig{
+		Timeout:  duration{10 * time.Minute},
+		Interval: duration{10 * time.Second},
+	},
+}
+
+type healthConfig struct {
+	// Components
+	Components componentHealthConfig `yaml:"components,omitempty" json:"components,omitempty"`
+	// Wait contains configuration concerning how the stand-by-period for the ecosystem to become healthy.
+	Wait waitHealthConfig `yaml:"wait,omitempty" json:"wait,omitempty"`
+}
+
+type componentHealthConfig struct {
+	Required []requiredComponent `yaml:"required,omitempty" json:"required,omitempty"`
+}
+
+type requiredComponent struct {
+	Name string `yaml:"name" json:"name"`
+}
+
+type waitHealthConfig struct {
+	Timeout  duration `yaml:"timeout,omitempty" json:"timeout,omitempty"`
+	Interval duration `yaml:"interval,omitempty" json:"interval,omitempty"`
+}
```

Comment:
you can move this types to the serializer too. Maybe even the default config.

Reply:
Oh wait, I didn't?

Reply:
I also started commenting and then forgor 💀